budgetportal/tests/test_provincial_infrastructure_projects.py

```diff
@@ -86,3 +87,88 @@ def test_upload_xlsx_for_prov_infra_projects(self):
         self.assertEqual(count, 11)
 
 
+class IRMReportSheetTestCase(TestCase):
+    def setUp(self):
+        dataset = Dataset()
+        dataset.headers = NORMAL_HEADERS + ["Project Contractor"] * 50
+        row = (
+            [1] * len(NORMAL_HEADERS)
+            + [
+                "Program Implementing Agent: DOPW",
+                "Program Implementing Agent: TEST",
+                "Main Contractor: AAAA",
+                "Principal Agent: BBBB",
+            ]
+            + [None] * 46
+        )
+        dataset.append(row=row)
+        self.report_sheet = IRMReportSheet(dataset)
+
+        dataset_with_other_keys = Dataset()
+        dataset_with_other_keys.headers = NORMAL_HEADERS + ["Project Contractor"] * 50
+        row_with_other_keys = (
+            [1] * len(NORMAL_HEADERS)
+            + [
+                "Service Provider: DOPW",
+                "Program Implementing Agent: TEST",
+                "Service Provider: AAAA",
+                "Principal Agent: BBBB",
+            ]
+            + [None] * 46
+        )
+        dataset_with_other_keys.append(row_with_other_keys)
+        self.sheet_with_other_keys = IRMReportSheet(dataset_with_other_keys)
```

Comment:
please split `sheet_with_other_keys` and `report_sheet` into two test classes - there's no value in them being the same class so separation will indicate that they are not connected.

Reply:
That will also improve data locality - where the test data is defined and checked will be close together, and not interleaved with other tests

Reply:
classes split:
https://github.com/vulekamali/datamanager/pull/189/commits/e8a09cf18378c191e260f74fd6ab5249b3ce4cbe